File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        leftSum = 0
        rightSum = sum(nums)

        for i in range(0, len(nums)):
            rightSum -= nums[i]
            if leftSum == rightSum: return i
            else:
                leftSum += nums[i]
        return -1

File: 0724-find-pivot-index/0724-find-pivot-index.go
func pivotIndex(nums []int) int {

    leftSum := 0
    rightSum := 0

    for _, num := range nums {
        rightSum += num
    }
    
    rightSum -= nums[0]

    if leftSum == rightSum {
        return 0
    }
    
    for i := 0; i < len(nums) - 1; i++ {
        rightSum -= nums[i+1]
        leftSum += nums[i]

        if leftSum == rightSum {
            return i + 1
        }
    }

    return -1
}